### PR TITLE
Refactor: Game

### PR DESCRIPTION
**Pull request by marcobiedermann**
Refactor game logic.

Again, there was nothing wrong with your code at all. I just wanted to give you some minor feedback on how you could decouple your UI from the business logic. This way, you could easily replace certain components (input, output, logic / model, view, controller).
Also I would recommend failing as early as possible. Adding some general validation at the very beginning keeps your other functions clean and pure.
Especially in React, you try to split your logic and UI layer (pure/stateless/dumb components from stateful/smart components).

You also don't have to accept this pull request or adjust your initial implementation, I just wanted to give you some recommendations and show *A* different solution. Not *THE* solution 😄 

Btw: I would recommend viewing the actual file: https://github.com/petekgithub/brainnest-js-assignment/blob/e2e53af9935568882bb4c22129c57c2d9698780e/script.js
Might be easier than comparing it to the previous

Keep on coding 🚀 💪

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -1,69 +1,101 @@
-function computerPlay() {
-  const choice = ['rock','paper','scissors'];
-  return choice[Math.floor(Math.random()*choice.length)]
+const choices = ['rock', 'paper', 'scissors'];
+
+function randomInteger(max, min = 0) {
```
Comment (marcobiedermann) [inline script.js:3]: add general utility function to generate a random integer within range

File: script.js [hunk 2]
```diff
@@ -1,69 +1,101 @@
...
+  return Math.floor(Math.random() * (max - min + 1)) + min;
 }
 
+function getRandomChoice() {
+  return choices[randomInteger(choices.length) - 1];
+}
 
-function playOneRound(playerSelection, computerSelection) {
+function sanitize(input) {
```
Comment (marcobiedermann) [inline script.js:11]: sanitize input before doing any further processing

File: script.js [hunk 3]
```diff
@@ -1,69 +1,101 @@
...
+}
 
-  if (playerSelection.toLowerCase() === 'paper') {
-    if(computerSelection.toLowerCase() === 'rock') { 
-      return `win! ${playerSelection} beats ${computerSelection}`;
-    }else {
-      return `lost! ${computerSelection} beats ${playerSelection}`;
-    }
+function getResult(player1, player2) {
+  const battleMap = {
```
Comment (marcobiedermann) [inline script.js:32]: I did not come up with a better name 🙈 
This stores all the main game logic in just one object.

File: script.js [hunk 4]
```diff
@@ -1,69 +1,101 @@
...
+    rock: 'scissors',
+    paper: 'rock',
+    scissors: 'paper'
+  };
+
+  if (player1 === player2) {
```
Comment (marcobiedermann) [inline script.js:38]: Return enum `'win' | 'lose' | 'tie'`. Alternatively `-1 | 0 | 1` but a string might be more explicit

File: script.js [hunk 5]
```diff
@@ -1,69 +1,101 @@
...
-      return `lost! ${computerSelection} beats ${playerSelection}`;
-    }
+  if (battleMap[player1] === player2) {
+    return 'win';
+  } else {
+    return 'lose';
   }
+}
 
+function getMessage(result, player1, player2) {
```
Comment (marcobiedermann) [inline script.js:49]: Just return a plain string. This way you can decide later how to display it

File: script.js [hunk 6]
```diff
@@ -1,69 +1,101 @@
...
+    tie: () => `It's a tie!`,
+    win: (player1, player2) => `You win! ${player1} beats ${player2}`,
+    lose: (player1, player2) => `You lose! ${player2} beats ${player1}`
+  };
+
+  return messages[result](player1, player2);
 }
 
-function game() {  
+function playRound(input) {
```
Comment (marcobiedermann) [inline script.js:59]: Clear step-by-step instructions.
Get input, sanitize it, validate it, handle game mechanics, get result

File: script.js [hunk 7]
```diff
@@ -1,69 +1,101 @@
...
 
-  let playerWins, computerWins, tie = 0;
+  return {
+    result,
+    message
+  };
+}
+
+function game() {
+  const scores = {
```
Comment (marcobiedermann) [inline script.js:75]: I rather prefer storing related information in an object instead of using separate variables

File: script.js [hunk 8]
```diff
@@ -1,69 +1,101 @@
...
 
-  for(let i = 0; i < 5; i++) {
-    const playerSelection = prompt("Please enter one of these: rock, paper or scissors")
-    const computerSelection = computerPlay();
+  const numberOfRounds = 5;
 
-    const winner = playOneRound(playerSelection,computerSelection);
-    console.log(winner);
+  for(let i = 0; i < numberOfRounds; i += 1) {
+    const input = prompt('Please enter one of these: rock, paper or scissors')
```
Comment (marcobiedermann) [inline script.js:84]: This way you could replace the `prompt` with an HTML `input` field. Your UI is decoupled

File: script.js [hunk 9]
```diff
@@ -1,69 +1,101 @@
...
+      scores.losses += 1
+    } else{
+      scores.ties += 1
     }
+
+    console.log(message);
   }
-  console.log(`You won: ${playerWins}, Computer won: ${computerWins} and Tie: ${tie} `)
+
+  console.log(`You won: ${scores.wins}, Computer won: ${scores.losses} and Tie: ${scores.ties} `)
```
Comment (marcobiedermann) [inline script.js:98]: Instead of logging, you could also render the message to a DOM node

File: script.js (hunk 1 above)
Comment (petekgithub) [reply script.js:3]: Do you mean, should I create a new function to generate a number integer and then call it?

**Review by petekgithub — APPROVED** [review]

**Pull request merged**